apis/nucleus/src/components/NebulaApp.jsx

```diff
@@ -23,7 +23,8 @@ const NebulaApp = forwardRef(({ initialContext, app }, ref) => {
 
   useImperativeHandle(ref, () => ({
     addComponent(component) {
-      setComponents([...components, component]);
+      const key = component.key || `${new Date().getTime()}_${Math.random().toString(36).substring(2)}`;
```

Comment:
You should be able to use `Date.now()` instead of `new Date().getTime()`.

Another option to consider would be using [crypto.randomUUID](https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID)